### PR TITLE
Switch core validation to text VUIDs

### PR DESCRIPTION
**Pull request by daveh-lunarg**
Remove all references to UNIQUE_VALIDATION_ERROR_CODE enums in core
validation and layer tests. Step 1 in a multi-step effort to remove
the enums and convert all validation code to work directly with the
spec's textual VUID strings.

Heads up: large touch footprint

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -7638,33 +7680,33 @@ static const char *vu_summary[] = {"Source or destination queue family must be i
...
-    VALIDATION_ERROR_0a000aca,  //  VUID-VkImageMemoryBarrier-image-01381 -- kSrcOrDstMustBeIgnore
-    VALIDATION_ERROR_0a000dcc,  //  VUID-VkImageMemoryBarrier-image-01766 -- kSpecialOrIgnoreOnly
-    VALIDATION_ERROR_0a000962,  //  VUID-VkImageMemoryBarrier-image-01201 -- kSrcIgnoreRequiresDstIgnore
-    VALIDATION_ERROR_0a000dd0,  //  VUID-VkImageMemoryBarrier-image-01768 -- kDstValidOrSpecialIfNotIgnore
-    VALIDATION_ERROR_0a000dce,  //  VUID-VkImageMemoryBarrier-image-01767 -- kSrcValidOrSpecialIfNotIgnore
-    VALIDATION_ERROR_0a00095e,  //  VUID-VkImageMemoryBarrier-image-01199 -- kSrcAndDestMustBeIgnore
-    VALIDATION_ERROR_0a000960,  //  VUID-VkImageMemoryBarrier-image-01200 -- kBothIgnoreOrBothValid
-    VALIDATION_ERROR_0a00096a,  //  VUID-VkImageMemoryBarrier-image-01205 -- kSubmitQueueMustMatchSrcOrDst
+static const std::string image_error_codes[] = {
+    "VUID-VkImageMemoryBarrier-image-01381",  //  VUID-VkImageMemoryBarrier-image-01381 -- kSrcOrDstMustBeIgnore
```
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp]: The VUID portion of the comment is no longer needed as the VUID's are the table entries now.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -1225,14 +1222,15 @@ static bool ValidatePipelineLocked(layer_data *dev_data, std::vector<std::unique
         PIPELINE_STATE *pBasePipeline = nullptr;
         if (!((pPipeline->graphicsPipelineCI.basePipelineHandle != VK_NULL_HANDLE) ^
               (pPipeline->graphicsPipelineCI.basePipelineIndex != -1))) {
-            // This check is a superset of VALIDATION_ERROR_096005a8 and VALIDATION_ERROR_096005aa
+            // This check is a superset of "VUID-VkGraphicsPipelineCreateInfo-flags-00724" and
+            // "VUID-VkGraphicsPipelineCreateInfo-flags-00725"
```
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp]: Do you want to also replace the old style enums with quoted equivalent strings, until we can either find the VUID text that applies or write a faux VUID "UNASSIGNED-<cmdOrStructName>-<fieldOrParam>-<pickANumber>"  as Mark suggested for the QFO Transfer "new vu's"

Comment (daveh-lunarg) [reply layers/core_validation.cpp]: I think so.  Not in this PR, though.

File: layers/buffer_validation.cpp [hunk 3]
```diff
@@ -2999,7 +3021,7 @@ static bool validate_usage_flags(layer_data *device_data, VkFlags actual, VkFlag
         correct_usage = ((actual & desired) != 0);
     }
     if (!correct_usage) {
-        if (msgCode == -1) {
+        if (msgCode == "VUID_Undefined") {
```
Comment (tobine) [inline layers/buffer_validation.cpp]: Should be const string, such as "kVUIDUndefined", defined in a central location that all uses refer to.

Comment (daveh-lunarg) [reply layers/buffer_validation.cpp]: Ok, I'll recover the enum from the string, for now.  I do think we want to rip off the band-aid as quickly as we can, but having a hybrid version for at least SDK cycle seems reasonable.

**Pull request merged**